[vcpkg-ci-ffmpeg, ffmpeg] Require CI pass on all triplets, fix dependencies

dg0yt opened the pull request.

dg0yt committed:
[vcpkg-ci-ffmpeg] Require pass on all CI triplets

dg0yt commented:
arm64-android: Need `-lc++`, but some deps still need work:
mpg123: -lstdc++
modplug: .../libstdc++
x265: .../libc++

cenit commented:
thanks. It was very difficult for me to accept to add failures for ffmpeg in my pr just because it was silently broken on master. I will make mine depend on this

dg0yt committed:
[ffmpeg] Use selected CMAKE_C_STANDARD_LIBARIES

For improving portability of interaction with cmake based dependencies
such as libwebp, apply standard libaries as cmake does: Use them for
linking, but do not output them in exported config.

dg0yt committed:
[vcpkg-ci-ffmpeg] Build a tool

dg0yt committed:
[libopenmpt] Export C++ linkage

dg0yt committed:
[libmodplug] Fix export of C++ linkage

dg0yt committed:
[ffmpeg] Don't enforce libstdc++ on libopenmpt

dg0yt committed:
[ffmpeg] Update FindFFMPEG.cmake.in

dg0yt committed:
[ffmpeg] Avoid redundant export fixup

WangWeiLin-MV commented on ports/libmodplug/portfile.cmake:
```diff
@@ -13,7 +13,7 @@ vcpkg_from_github(
         002-detect_sinf.patch
         003-use-static-cast-for-ctype.patch
-        004-export-pkgconfig.patch  # https://github.com/Konstanty/libmodplug/pull/59
+        004-export-pkgconfig.patch
```
Since the content of the patch has changed, it is recommended to commit it to upstream for tracking.

dg0yt replied:
Basically yes. OTOH upstream didn'r accept the original PR in three years.

WangWeiLin-MV reviewed (approved):
The ports `ffmpeg`, `libmodplug`, `libopenmpt` installation tests pass with the following triplets:
* x64-windows
